kolibri/core/apps.py

```diff
@@ -84,3 +88,51 @@ def activate_pragmas_on_start():
             # at the cost of a slight penalty to all reads.
             cursor.execute(START_PRAGMAS)
             connection.close()
+
+    # flake8: noqa: C901
+    @staticmethod
+    def check_redis_settings():
+        """
+        Check that Redis settings are sensible, and use the lower level Redis client to make updates
+        if we are configured to do so, and if we should, otherwise make some logging noise.
+        """
+        if OPTIONS["Cache"]["CACHE_BACKEND"] != "redis":
+            return
+        config_use_conf = OPTIONS["Cache"]["CACHE_REDIS_USE_CONF"]
+        config_maxmemory = OPTIONS["Cache"]["CACHE_REDIS_MAXMEMORY"]
+        config_maxmemory_policy = OPTIONS["Cache"]["CACHE_REDIS_MAXMEMORY_POLICY"]
+
+        try:
+            # see redis_cache.backends.single.RedisCache
+            helper = RedisSettingsHelper(process_cache.get_master_client())
+
+            # default setting is "0", or no limit
+            maxmemory = helper.get_maxmemory()
+            if config_use_conf:
+                if maxmemory == 0:
+                    logger.warning("Redis is configured without a maximum memory size.")
+            elif maxmemory != config_maxmemory:
+                helper.set_maxmemory(config_maxmemory)
+
+            # default setting is "noeviction"
+            maxmemory_policy = helper.get_maxmemory_policy()
+            if config_use_conf:
```

Comment:
same as above, it should be `config_maxmemory_policy` instead of `config_use_conf`. We should not change the redis maxmemory policy unless it's explicitly requested.